contract_manager/src/contracts/evm.ts

```diff
@@ -592,19 +592,29 @@ export class EvmEntropyContract extends Storable {
   async requestRandomness(
     userRandomNumber: string,
     provider: string,
-    senderPrivateKey: PrivateKey
+    senderPrivateKey: PrivateKey,
+    withCallback?: boolean
   ) {
     const web3 = new Web3(this.chain.getRpcUrl());
     const userCommitment = web3.utils.keccak256(userRandomNumber);
     const contract = new web3.eth.Contract(EXTENDED_ENTROPY_ABI, this.address);
     const fee = await contract.methods.getFee(provider).call();
     const { address } = web3.eth.accounts.wallet.add(senderPrivateKey);
-    const useBlockHash = false;
-    const transactionObject = contract.methods.request(
-      provider,
-      userCommitment,
-      useBlockHash
-    );
+
+    let transactionObject;
+    if (withCallback) {
+      transactionObject = contract.methods.requestWithCallback(
+        provider,
+        userCommitment
+      );
+    } else {
+      transactionObject = contract.methods.request(
+        provider,
+        userCommitment,
+        false
+      ); // false for useBlockHash
```

Comment:
I think the old way of doing this `const useBlockHash = false;` was more readable. You can define that const in the `else` condition alone.